consul/resource_consul_config_entry_test.go

```diff
@@ -0,0 +1,278 @@
+					resource.TestCheckResourceAttr("consul_config_entry.service_router", "config_json", "{\"Routes\":[{\"Destination\":{\"Service\":\"admin\"},\"Match\":{\"HTTP\":{\"PathPrefix\":\"/admin\"}}}]}"),
+				),
+			},
+			{
+				Config: TestAccConsulConfigEntry_ServiceSplitter,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("consul_config_entry.service_splitter", "name", "web"),
+					resource.TestCheckResourceAttr("consul_config_entry.service_splitter", "kind", "service-splitter"),
+					resource.TestCheckResourceAttr("consul_config_entry.service_splitter", "config_json", "{\"Splits\":[{\"ServiceSubset\":\"v1\",\"Weight\":90},{\"ServiceSubset\":\"v2\",\"Weight\":10}]}"),
+				),
+			},
+			{
+				Config: TestAccConsulConfigEntry_ServiceResolver,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("consul_config_entry.service_resolver", "name", "web"),
+					resource.TestCheckResourceAttr("consul_config_entry.service_resolver", "kind", "service-resolver"),
+					resource.TestCheckResourceAttr("consul_config_entry.service_resolver", "config_json", "{\"DefaultSubset\":\"v1\",\"Subsets\":{\"v1\":{\"Filter\":\"Service.Meta.version == v1\"},\"v2\":{\"Filter\":\"Service.Meta.version == v2\"}}}"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccConsulConfigEntry_Errors(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() {},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccConsulConfigEntry_ProxyDefaultsWrongName,
+				ExpectError: regexp.MustCompile("Provider produced inconsistent result after apply: When applying changes to consul_config_entry.foo, provider \"consul\" produced an unexpected new value for was present, but now absent."),
+			},
+		},
+	})
+}
+
+const testAccConsulConfigEntry_ServiceDefaults = `
+resource "consul_config_entry" "foo" {
+	name = "foo"
+	kind = "service-defaults"
+
+	config_json = jsonencode({
+		MeshGateway = {}
+		Protocol    = "https"
+	})
+}
+`
+
+const testAccConsulConfigEntry_ServiceDefaultsOptionalField = `
+resource "consul_config_entry" "foo" {
+	name = "foo"
+	kind = "service-defaults"
+
+	config_json = jsonencode({
+		Protocol    = "https"
+	})
+}
+`
+
+const testAccConsulConfigEntry_ServiceDefaultsExtraField = `
+resource "consul_config_entry" "foo" {
+	name = "foo"
+	kind = "service-defaults"
+
+	config_json = jsonencode({
+		ThisFieldDoesNotExists = true
+		Protocol               = "https"
+	})
+}
+`
+
+const testAccConsulConfigEntry_ProxyDefaults = `
+resource "consul_config_entry" "foo" {
+	name = "global"
+	kind = "proxy-defaults"
+
+	config_json = jsonencode({
+		MeshGateway = {}
+		Config      = {
+			foo = "bar"
+		}
+	})
+}
+`
+
+const TestAccConsulConfigEntry_ServiceRouter = `
+resource "consul_config_entry" "web" {
+	name = "web"
+	kind = "service-defaults"
+
+	config_json = jsonencode({
+		MeshGateway = {}
```

Comment:
I see that we test both with and without this default now so I'm happy it passes both ways but as a nit pick I'd personally leave it out of all but the one test case we actually want to check it works as that reflects typical usage better IMO.